### PR TITLE
Change siddhi import sample view and add sample search feature.

### PR DESCRIPTION
**Pull request by kts-desilva**
## Purpose
Change the current siddhi import sample dialog view from jsTree to a table with the sample description included and add the search feature for the samples.

## Goals
User should be able to import the siddhi samples via a table view.
User should be able to view the siddhi samples with their descriptions.
User should be able to search the siddhi samples.

## Approach
Converting the jsTree view into table view.
Adding the descriptions along with the siddhi samples.
Adding the search bar to filter the siddhi samples by the sample title and description.

The new siddhi import sample dialog view is as follows.

screenshot from 2018-11-21 08-51-14

### CODE REVIEW

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java [hunk 1]
```diff
@@ -931,4 +948,37 @@ protected void registerConfigProvider(ConfigProvider configProvider) {
     protected void unregisterConfigProvider(ConfigProvider configProvider) {
         this.configProvider = null;
     }
+
+    protected void getSampleFiles() {
```
Comment (ramindu90) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java]: @kts-desilva can we change this name to "loadSampleFiles"

Comment (kts-desilva) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java]: Sure. Fixed in 43b4465c20aa7a2c50581e2401f9a2ac6fc7ebf0

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/open-sample-file-dialog.js [hunk 2]
```diff
@@ -128,66 +124,54 @@ define(['require', 'lodash','jquery', 'log', 'backbone', 'file_browser', 'worksp
...
-                });
-
-                sampleFileOpen.find("button").filter("#openButton").click(function () {
+                var treeContainer  = sampleFileOpen.find("div").filter("#sampleTable");
+                var bodyUlSampleContent = $('<ul class="recent-files clearfix"></ul>');
+                bodyUlSampleContent.attr('id', "sampleList");
+                var browserStorage = app.browserStorage;
+                var workspaceServiceURL = app.config.services.workspace.endpoint;
+                var getSampleDescServiceURL = workspaceServiceURL + "/listFiles/samples/descriptions";
+                var samplesWithDes={};
```
Comment (ramindu90) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/open-sample-file-dialog.js]: Could you please format this JS code? there are several places without formatting

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/sample-view/sample-view.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        var config = this._config;
+        var errMsg;
+        if (!_.has(config, 'parentContainer')) {
+            errMsg = 'unable to find configuration for parentContainer';
+            log.error(errMsg);
+            throw errMsg;
+        }
+        // get parent container which is innerSamples div
+        var parentContainer = _.get(config, 'parentContainer');
+
```
Comment (ramindu90) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/sample-view/sample-view.js]: @kts-desilva could you please remove unnecessary spaces within methods

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java [hunk 4]
```diff
@@ -931,4 +947,37 @@ protected void registerConfigProvider(ConfigProvider configProvider) {
...
+            for (java.nio.file.Path path : collect) {
+                String fileContent = new String(Files.readAllBytes(path), Charset.defaultCharset());
+                Matcher matcher = pattern.matcher(fileContent);
+                String descriptionText = "";
+                if (matcher.find()) {
+                    String description = matcher.group();
+                    descriptionText = description.substring(description.indexOf("(") + 1, description.lastIndexOf(")"));
+                }
+                java.nio.file.Path relativeSamplePath=pathLocation.relativize(path);
+                sampleMap.put(relativeSamplePath.toString(),descriptionText);
```
Comment (minudika) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java]: please check and fix the checkstyle issues.

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/open-sample-file-dialog.js [hunk 5]
```diff
@@ -261,6 +245,64 @@ define(['require', 'lodash','jquery', 'log', 'backbone', 'file_browser', 'worksp
...
+                            openFileWizardError.show();
+                        }
+                    });
+                };
+
+                function searchSample(sampleContent,searchText){
+                    var filter = searchText.toUpperCase();
+                    var sampleElements = sampleContent[0].getElementsByTagName("li");
+                    var noResultsElement  = sampleFileOpen.find("div").filter("#noResults");
+                    var unmatchedCount=0;
```
Comment (ramindu90) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/open-sample-file-dialog.js]: @kts-desilva  Can you format this code? Please check other files as well.

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java [hunk 6]
```diff
@@ -931,4 +948,37 @@ protected void registerConfigProvider(ConfigProvider configProvider) {
...
+                    .collect(Collectors.toList());
+            for (java.nio.file.Path path : collect) {
+                String fileContent = new String(Files.readAllBytes(path), Charset.defaultCharset());
+                Matcher matcher = pattern.matcher(fileContent);
+                String descriptionText = "";
+                if (matcher.find()) {
+                    String description = matcher.group();
+                    descriptionText = description.substring(description.indexOf("(") + 1, description.lastIndexOf(")"));
+                }
+                java.nio.file.Path relativeSamplePath=pathLocation.relativize(path);
```
Comment (ramindu90) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java]: formatting issue

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/open-sample-file-dialog.js (hunk 2 above)
Comment (kts-desilva) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/open-sample-file-dialog.js]: Sure. Fixed in f228ff475e8dd19e1ef997ba24528c91a61ff51e

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/sample-view/sample-view.js (hunk 3 above)
Comment (kts-desilva) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/sample-view/sample-view.js]: Sure. Fixed in f228ff475e8dd19e1ef997ba24528c91a61ff51e

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/open-sample-file-dialog.js (hunk 5 above)
Comment (kts-desilva) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/open-sample-file-dialog.js]: Sure. Fixed in f228ff475e8dd19e1ef997ba24528c91a61ff51e

**Comment by Niveathika92** [comment]
@kts-desilva Do we need to improve samples? In the screenshot some have descriptions while others dont

**Comment by kts-desilva** [comment]
@Niveathika92 Yes. Since it is required to show the samples along with their descriptions.

**Comment by erangatl** [comment]
Yes we need to add descriptions for all, also its better if we can have a properly formatted title for the samples.

**Comment by Niveathika92** [comment]
Dismissing review as the changes requested have to be done at samples level

**Pull request merged**